packages/@aws-cdk/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -0,0 +1,195 @@
+      stateMachineName = (this.stateMachine.node.defaultChild as sfn.CfnStateMachine).stateMachineName;
+    } else {
+      //imported state machine
+      stateMachineName = 'StateMachine-' + (String(this.stateMachine.stack.node.addr).substring(0, 8));
+    }
+
+    let deploymentToken;
+
+    if (stateMachineName !== undefined && !Token.isUnresolved(stateMachineName)) {
+      deploymentToken = JSON.stringify({ stateMachineName });
+    }
+    return {
+      ...bindResult,
+      deploymentToken,
+    };
+
+  }
+}
+
+function integrationResponse() {
+  const errorResponse = [
+    {
+      /**
+       * Specifies the regular expression (regex) pattern used to choose
+       * an integration response based on the response from the back end.
+       * In this case it will match all '4XX' HTTP Errors
+       */
+      selectionPattern: '4\\d{2}',
+      statusCode: '400',
+      responseTemplates: {
+        'application/json': `{
+            "error": "Bad input!"
+          }`,
+      },
+    },
+    {
+      /**
+       * Match all '5XX' HTTP Errors
+       */
+      selectionPattern: '5\\d{2}',
+      statusCode: '500',
+      responseTemplates: {
+        'application/json': '"error": $input.path(\'$.error\')',
+      },
+    },
+  ];
+
+  const integResponse = [
+    {
+      statusCode: '200',
+      responseTemplates: {
+        'application/json': `#set($inputRoot = $input.path('$'))
+                #if($input.path('$.status').toString().equals("FAILED"))
+                    #set($context.responseOverride.status = 500)
+                    { 
+                      "error": "$input.path('$.error')",
+                      "cause": "$input.path('$.cause')"
+                    }
+                #else
+                    $input.path('$.output')
+                #end`,
+      },
+    },
+    ...errorResponse,
+  ];
+
+  return integResponse;
+}
+
+function requestTemplates(stateMachine: sfn.IStateMachine, includeRequestContext: boolean | undefined) {
+  const templateStr = templateString(stateMachine, includeRequestContext);
+
+  const requestTemplate: { [contentType:string] : string } =
+    {
+      'application/json': templateStr,
+    };
+
+  return requestTemplate;
+}
+
+function templateString(stateMachine: sfn.IStateMachine, includeRequestContext: boolean | undefined): string {
+  let templateStr: string;
+  const requestContextStr = requestContext();
+
+  const search = '"';
+  const replaceWith = '\\"';
+  if (typeof includeRequestContext === 'boolean' && includeRequestContext === true) {
+    templateStr = `
+    #set($allParams = $input.params())
+    {
+      "input": "{${(requestContextStr.split(search).join(replaceWith))}}",
+      "stateMachineArn": "${stateMachine.stateMachineArn}"
+    }`;
+  } else {
+    templateStr = `
+        #set($inputRoot = $input.path('$')) {
+            "input": "$util.escapeJavaScript($input.json('$'))",
+            "stateMachineArn": "${stateMachine.stateMachineArn}"
+          }`;
+  }
```

Comment:
Quite hard to understand because of the interlacing of vtl and typescript.

Can you clean this up and make this easy to understand with comments.